### PR TITLE
fix(#1100): empty file resource returns true on exist

### PR DESCRIPTION
**Pull request by tschlat**
Fixes issue #1100:
An UrlResource pointing to an existing but empty file must return true on exist.

Also added a couple of tests.

### CODE REVIEW

File: core/citrus-api/src/main/java/org/citrusframework/spi/Resources.java [hunk 1]
```diff
@@ -263,11 +268,27 @@ public String getLocation() {
 
         @Override
         public boolean exists() {
+
+
```
Comment (bbortt) [inline core/citrus-api/src/main/java/org/citrusframework/spi/Resources.java:272]:
```suggestion
```

please do also update the copyright year to `2023-2024` on top of this class.

File: core/citrus-api/src/test/java/org/citrusframework/spi/ResourcesTest.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.spi.Resources.ByteArrayResource;
+import org.citrusframework.spi.Resources.ClasspathResource;
+import org.citrusframework.spi.Resources.FileSystemResource;
+import org.citrusframework.spi.Resources.UrlResource;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class ResourcesTest {
```
Comment (bbortt) [inline core/citrus-api/src/test/java/org/citrusframework/spi/ResourcesTest.java:31]: this file is not properly formatted.

**Review by bbortt — CHANGES_REQUESTED** [review]
thank you very much! looks good to me in general, only some nitpicks.

**Review by tschlat — COMMENTED** [review]
Done

**Comment by bbortt** [comment]
> Done

well thank you!

**Review by bbortt — APPROVED** [review]

**Review by christophd — APPROVED** [review]
Awesome!

**Pull request merged**